src/test/java/uk/gov/companieshouse/service/DissolutionPatcherTest.java

```diff
@@ -112,4 +115,19 @@ public void patch_doesNotUpdateStatus_ifNotAllDirectorHaveApproved() {
                 dissolutionCaptor.getValue().getData().getApplication().getStatus()
         );
     }
+
+    @Test
+    public void patch_updatesDissolutionWithPaymentInformation_savesInDatabase() {
+        PaymentPatchRequest data = generatePaymentPatchRequest();
+
+        when(repository.findByCompanyNumber(COMPANY_NUMBER)).thenReturn(java.util.Optional.of(dissolution));
+
+        patcher.updatePaymentStatus(data.getPaymentReference(), data.getPaidAt(), COMPANY_NUMBER);
+        verify(repository).save(dissolutionCaptor.capture());
+
+        assertEquals(ApplicationStatus.PAID, dissolutionCaptor.getValue().getData().getApplication().getStatus());
+        assertEquals(data.getPaymentReference(), dissolutionCaptor.getValue().getPaymentInformation().getReference());
+        assertEquals(data.getPaidAt(), Timestamp.valueOf(dissolutionCaptor.getValue().getPaymentInformation().getDateTime()));
+        assertEquals(PaymentMethod.CREDIT_CARD, dissolutionCaptor.getValue().getPaymentInformation().getMethod());
```

Comment:
yeah I think this corrupts the purpose of the service. Ideally this would be:
```
assertEquals(mockPaymentInfo, dissolutionCaptor.getValue().getPaymentInformation());
```
Then the mapper tests would verify the individual fields were set correctly.